js/ui/payg.js

```diff
@@ -465,3 +472,67 @@ function timeToString(seconds) {
     let hoursStr = Gettext.ngettext('%s hour', '%s hours', hoursPast).format(hoursPast);
     return '%s %s'.format(daysStr, hoursStr);
 }
+
+var PaygNotifier = GObject.registerClass(
+class PaygNotifier extends GObject.Object {
+    _init() {
+        super._init();
+
+        this._notification = null;
+    }
+
+    notify(secondsLeft) {
+        // Only notify when in an regular session, not in GDM or initial-setup.
```

Comment:
There's a mismatch between the comment and the code, since we're not checking for the `gdm` or `initial-setup` modes. And similarly in `js/ui/status/automaticUpdates.js`.